### PR TITLE
Fix rest server issue

### PR DESCRIPTION
**Pull request by freddyli7**
## Description

- Fix the rest server issue
- Added tx query, tx generate, encode, decode, broadcast endpoints.

### CODE REVIEW

File: x/chainlink/client/rest/queryFeed.go [hunk 1]
```diff
@@ -11,31 +10,18 @@ import (
 	"github.com/gorilla/mux"
 )
 
-func listFeedDataHandler(clientCtx client.Context) http.HandlerFunc {
+func listFeedDataByFeedIdHandler(clientCtx client.Context) http.HandlerFunc {
```
Comment (araskachoi) [inline x/chainlink/client/rest/queryFeed.go:13]: nice! i like the new naming

File: x/chainlink/client/rest/queryFeed.go [hunk 2]
```diff
@@ -11,31 +10,18 @@ import (
...
-		fmt.Println("???????", feedId)
-
-		params := &types.QueryFeedDataRequest{
-			FeedId:     feedId,
-			Pagination: nil,
-		}
-
-		queryClient := types.NewQueryClient(clientCtx)
-		res, err := queryClient.FeedDataByID(context.Background(), params)
+		res, height, err := clientCtx.QueryWithData(fmt.Sprintf("custom/%s/%s/%s", types.StoreKey, types.QueryFeedData, feedId), nil)
```
Comment (araskachoi) [inline x/chainlink/client/rest/queryFeed.go:18]: this will eventually include the `roundid` too, right?

Comment (freddyli7) [reply x/chainlink/client/rest/queryFeed.go:18]: yes, we need to impl roundID first and it will be param in query later

Comment (araskachoi) [reply x/chainlink/client/rest/queryFeed.go:18]: idk, this doesnt need to be changed right now, but eventually i think

File: x/chainlink/client/rest/rest.go [hunk 3]
```diff
@@ -18,10 +19,18 @@ func RegisterRoutes(clientCtx client.Context, r *mux.Router) {
...
-func registerTxHandlers(clientCtx client.Context, r *mux.Router) {
-	r.Handle("chainlink/feed/data", createFeedHandler(clientCtx)).Methods(MethodPOST)
+func registerTxHandlers(clientCtx client.Context, rtr *mux.Router) {
+	r := clientrest.WithHTTPDeprecationHeaders(rtr)
+
+	r.HandleFunc("/txs/encode", authrest.EncodeTxRequestHandlerFn(clientCtx)).Methods(MethodPOST)
+	r.HandleFunc("/txs/decode", authrest.DecodeTxRequestHandlerFn(clientCtx)).Methods(MethodPOST)
+	r.HandleFunc("/txs", authrest.BroadcastTxRequest(clientCtx)).Methods(MethodPOST)
+
+	r.Handle("/chainlink/feed/data", submitFeedDataHandler(clientCtx)).Methods(MethodPUT)
```
Comment (araskachoi) [inline x/chainlink/client/rest/rest.go:35]: why do we need PUT instead of POST here? jw

File: x/chainlink/client/rest/queryFeed.go (hunk 2 above)
Comment (freddyli7) [reply x/chainlink/client/rest/queryFeed.go:18]: correct, that is the path after we have roundID

File: x/chainlink/client/rest/txFeed.go [hunk 4]
```diff
@@ -16,26 +17,31 @@ type createFeedRequest struct {
...
 			return
 		}
-
 		baseReq := req.BaseReq.Sanitize()
 		if !baseReq.ValidateBasic(w) {
 			return
 		}
 
-		submitter, err := sdk.AccAddressFromBech32(req.Submitter)
+		submitter, err := sdk.AccAddressFromBech32(baseReq.From)
```
Comment (araskachoi) [inline x/chainlink/client/rest/txFeed.go:32]: some of these params will need to change with the fixes to the feedData Structure that will be coming in from a different PR.

Comment (araskachoi) [reply x/chainlink/client/rest/txFeed.go:32]: but again, not to be done right now but eventually! ^^

File: x/chainlink/client/rest/rest.go (hunk 3 above)
Comment (freddyli7) [reply x/chainlink/client/rest/rest.go:35]: PUT is used for submit, but it doesnt really `submit` data to store, instead it just returns a Std.Msg object, caller will need to use `broadcast` which is a POST to modify store

**Review by araskachoi — APPROVED** [review]
lgtm, left some comments that may need to be accounted for in the future. but currently, looks good to merge.
